client/src/main/java/org/apache/celeborn/client/read/CelebornInputStream.java

```diff
@@ -721,6 +740,36 @@ public synchronized void close() {
       }
     }
 
+    void validateIntegrity() throws IOException {
+      if (integrityChecked || !shuffleIntegrityCheckEnabled) {
+        return;
+      }
+
+      if (readSkewPartitionWithoutMapRange) {
+        shuffleClient.readReducerPartitionEnd(
```

Comment:
The current implementation mixes the concepts of numberOfSubPartitions/currentIndexOfSubPartition and startMapIndex/endMapIndex across different method signatures, such as readReducerPartitionEnd(int startMapIndex, int endMapIndex) and readPartition(int startMapIndex, int endMapIndex). Meanwhile, CelebornInputStreamImpl utilizes both. To ensure consistency, IMO we should use (int startMapIndex, int endMapIndex), as seen in CelebornShuffleReader.

Reply:
The existing Celeborn implementation in `CelebornInputStream` explicitly sets the `startMapIndex` and `endMapIndex` to -1 in the case of `skewPartitionHandlingWithoutMapRange`. That is why i left the existing convention alone.
Would you know why the original implementation chose to set the values to -1?

Reply:
> I propose we keep this as is for now. If required, we can always make a smaller change in a follow up PR. Will make it easier to reason about as well

Agree, this can be done in next prs